media/js/src/forms/SelectionForm.jsx

```diff
@@ -0,0 +1,228 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Creatable from 'react-select/creatable';
+import Select from 'react-select';
+import Button from 'react-bootstrap/Button';
+import Form from 'react-bootstrap/Form';
+
+import {
+    tagsToReactSelect, termsToReactSelect, termsToReactSelectValues
+} from '../utils';
+import {hasSelection} from '../openlayersUtils';
+
+const noSelectionError =
+      'Please select a portion of the media using the selections tools.';
+
+const reactSelectStyles = {
+    container: (provided, state) => ({
+        ...provided,
+        padding: 0,
+        height: 'fit-content',
+    }),
+    control: (provided, state) => ({
+        ...provided,
+        borderWidth: 0,
+        minHeight: 'fit-content',
+        height: 'fit-content'
+    }),
+    singleValue: (provided, state) => ({
+        ...provided,
+        top: '45%'
+    }),
+    placeholder: (provided, state) => ({
+        ...provided,
+        top: '45%'
+    })
+};
+
+export default class SelectionForm extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            validated: false,
+
+            title: this.props.selection ? this.props.selection.title : '',
+            tags: this.props.selection ? this.props.selection.tags : '',
+            terms: this.props.selection ? this.props.selection.terms : '',
+            notes: this.props.selection ? this.props.selection.body : ''
+        };
+
+        this.tagsRef = React.createRef();
+        this.termsRef = React.createRef();
+        this.titleFieldRef = React.createRef();
+
+        this.handleTitleChange = this.handleTitleChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    handleTitleChange(e) {
+        if (!e) {
+            return;
+        }
+
+        this.setState({title: e.target.value});
+    }
+
+    handleSubmit(e) {
+        // Before checking the HTML5 form's validity, do a custom
+        // validation check to make sure the user has made a
+        // selection.
+        if (
+            (
+                this.props.type === 'image' &&
+                    !hasSelection(this.props.selectionSource)
+            ) || (
+                this.props.type === 'video' &&
+                    !this.props.selectionStartTime &&
+                    !this.props.selectionEndTime
+            )
+        ) {
+            e.preventDefault();
+            e.stopPropagation();
+            this.props.onShowValidationError(noSelectionError);
```

Comment:
For video editing, the code is falling into this path for valid selection start & end ranges. I'm not quickly seeing why this is the case.

Reply:
This is resolved - the video selection props weren't getting passed in to this new form.